### ISSUE TITLE
We want regex pattern rewrite

### ISSUE THREAD

**Issue by FrostyLeaf** [datawire/ambassador#1162]
Hope ambassador to support some rules like below:

```js
      getambassador.io/config: |
        ---
        apiVersion: ambassador/v0
        kind:  Mapping
        name:  regex_pattern_mapping
        prefix_regex: true
        prefix: /svc/([\w-]+)/([\w-]+)/
        service: $2.$1:80
        host_rewrite: $2.$1
        ---
        apiVersion: ambassador/v0
        kind:  Mapping
        name:  regex_pattern_mapping2
        prefix_regex: true
        prefix: /svc/{namespace}/{svcname}/
        service: $namespace.$svcname:80
        host_rewrite: $namespace.$svcname
```

**Comment by richarddli** [comment]
PR would be great. (Or at least, an investigation into whether or not this is a supported Envoy feature.)

**Comment by FrostyLeaf** [comment]
This looks good:

https://github.com/envoyproxy/envoy/issues/2092#issue-275899030

**Comment by harindaka** [comment]
not stale

**Comment by kphelps** [comment]
It looks like envoy support for this recently landed https://github.com/envoyproxy/envoy/pull/10050. Regex-based path rewriting would be really helpful for us, too.

**Comment by marcosArruda** [comment]
I would implement this if I know how to do it :). May I help in any form?

**Comment by michaelfindlater** [comment]
This would be really helpful.

**Comment by alphashr** [comment]
FYI the Ambassador's `regex_rewrite` feature is just [released today](https://blog.getambassador.io/the-ambassador-edge-stack-ambassador-api-gateway-1-5-now-available-436d62a35839). You can read more about it [here](https://www.getambassador.io/docs/latest/topics/using/rewrites/) in docs.

**Comment by FrostyLeaf** [comment]
@alphashr, cool! Thanks for reminding me. This can be helpful.
